pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -0,0 +1,323 @@
+                zero_optimization,
+                zero_allow_untested_optimizer,
+                stage=stage,
+                cpu_offload=cpu_offload,
+                contiguous_gradients=contiguous_gradients,
+                overlap_comm=overlap_comm,
+                allgather_partitions=allgather_partitions,
+                reduce_scatter=reduce_scatter,
+                allgather_bucket_size=allgather_bucket_size,
+                reduce_bucket_size=reduce_bucket_size
+            )
+        self._config_initialized = False
+        deepspeed.utils.logging.logger.setLevel(logging_level)
+
+    def _load_config(self, config):
+        if config is None and self.DEEPSPEED_ENV_VAR in os.environ:
+            rank_zero_info(f"Loading DeepSpeed config from set {self.DEEPSPEED_ENV_VAR} environment variable")
+            config = os.environ[self.DEEPSPEED_ENV_VAR]
+        if isinstance(config, str) or isinstance(config, Path):
+            if os.path.exists(config):
+                with open(config) as f:
+                    config = json.load(f)
+            else:
+                raise MisconfigurationException(
+                    f"You passed in a path to a DeepSpeed config but the path does not exist: {config}"
+                )
+        return config
+
+    def pre_dispatch(self):
+        self.set_world_ranks()
+        self.init_ddp_connection(self.global_rank, self.world_size)
+
+        self.init_deepspeed()
+
+        # set warning rank
+        rank_zero_only.rank = self.global_rank
+
+        # set the ranks and devices
+        self.dist.rank = self.global_rank
+        self.dist.device = self.root_device
+        self.barrier()
+
+    def init_deepspeed(self):
+        if not self._config_initialized:
+            self._format_config()
+            self._config_initialized = True
+
+        precision = self.lightning_module.trainer.accelerator_backend.precision
+        model = LightningDeepSpeedModule(pl_module=self.model, precision=precision)
+
+        if self.lightning_module.trainer.training:
+            self._initialize_deepspeed_train(model)
+        else:
+            self._initialize_deepspeed_inference(model)
+
+    def _init_scheduler_optimizer(self):
+        optimizers, schedulers, optimizer_frequencies = self.lightning_module.trainer.init_optimizers(
+            self.lightning_module
+        )
+        if (len(optimizers) != 1) or len(schedulers) > 1:
+            raise MisconfigurationException(
+                "DeepSpeed currently only supports single optimizer, single optional scheduler."
+            )
+        scheduler = schedulers[0]['scheduler'] if len(schedulers) == 1 else None
+        optimizer = optimizers[0]
+        return optimizer, scheduler, optimizer_frequencies
+
+    def _initialize_deepspeed_train(self, model):
+        optimizer, lightning_scheduler, optimizer_frequencies = None, None, None
+        if "optimizer" not in self.config:
```

Comment:
Could the user specify optimiser and not scheduler or vide-versa ?

Reply:
Yes they could! 

I plan to do a few followup PRs to ease DeepSpeed integration in these cases, these are not super essential but very valid points :)